### ISSUE TITLE
Error loading service com/fasterxml/jackson/datatype/jsr310/JavaTimeModule

### ISSUE THREAD

**Issue by zdegner** [micronaut-projects/micronaut-openapi#160]
I'm wondering if anyone has any ideas how to work past this compilation error:

```
warning: TypeElementVisitor [io.micronaut.openapi.visitor.OpenApiApplicationVisitor] will be ignored due to loading error: Error loading service [io.micronaut.openapi.visitor.OpenApiApplicationVisitor]: com/fasterxml/jackson/datatype/jsr310/JavaTimeModule

warning: TypeElementVisitor [io.micronaut.openapi.visitor.OpenApiControllerVisitor] will be ignored due to loading error: Error loading service [io.micronaut.openapi.visitor.OpenApiControllerVisitor]: com/fasterxml/jackson/datatype/jsr310/JavaTimeModule
```

This error started occurring after updating to the latest version of Micronaut (1.2.8 -> 1.3.2). It prevents compilation of the application. However, the error goes away if I remove the annotationProcessor "io.micronaut.configuration:micronaut-openapi".

I've confirmed the jackson JSR310 library is on the classpath:

```
./gradlew -q dependencyInsight --dependency jackson-datatype-jsr310

com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.10.1 (selected by rule)
...

com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.10.1
\--- io.micronaut:micronaut-runtime:1.3.2
...
```

Any ideas how I could get more detailed logs?

**Comment by croudet** [comment]
Try to run gradle with `--stacktrace`

**Comment by zdegner** [comment]
Adding `--stacktrace` does not provide any additional information in this case. It print out an exception indicating only that compiling failed. It seems that the detailed inner exception is swallowed by TypeElementVisitorProcessor:

https://github.com/micronaut-projects/micronaut-core/blob/f1480baba9d657d76479846871b61557019c1417/inject-java/src/main/java/io/micronaut/annotation/processing/TypeElementVisitorProcessor.java#L131-L133

**Comment by croudet** [comment]
Can you set up an example and upload it to github?

**Comment by sven-kanoshin** [comment]
Hello.

We had the same issue. Spent quite a few hours trying to find the cause and hankfully the mentioned source code lines helped me to finally find the place to debug.
For our code, it was a NoClassDefFound for the reason that "jackson-datatype-jsr310" was not on the "annotationProcessor" profile. It was on the classpath, as per OP, but that wasn't sufficient. (Since I think the annotation processing has a different classloader consisting only of the libaries from the annotationProcessor configuration). Adding the dependency solved the issue for us. Maybe this could help the OP as well.

From my part 2 suggestions:
*) The NoDefFound is eaten by the different layers, perhaps it would be possible to make it more visible ? As can be seen from the OP's log lines, it currently is a bit... unhelpful, unless you already know what's going on.
*) The current issue is presented in the output as a warning, however as the 'OpenApiApplicationVisitor' is not running, the end effect is that no swagger is generated, which for me sounds rather more of an 'error' than warning. Perhaps there is a chance to improve that situation as well ? For our case, noone was looking for the up-to-date doc, ourselves included, so I stumbled upon the missing file accidentally.

Thank you for your work

**Comment by zdegner** [comment]
This appears to be resolved in Micronaut 2.0.0. Closing this issue.
